hooks/useLocaleUrl.ts

```diff
@@ -0,0 +1,16 @@
+'use client'
+
+export function generateNewUrl(pathname: string, language: string): string {
```

Comment:
Please use the `use` naming convention for hooks, so `useGenerateNewUrl` and `useCurrentLocale` or something.

I also think `generateNewUrl` does not describe very well what the hook is actually doing, perhaps take a look at `useLocalizedRoutes` and see if you can incorporate this logic into there because we seem to be creating a bunch of hooks that do basically the same but slightly different and this can get confusing for other contributors.

Reply:
Yes, I think I will be able to utilize the `usePathData` hook for all the information that I require for re direct (after a modification for providing the entire pathName as well).

Reply:
the function inside the generateNewUrl hook is very specific and I think having it inside the component file itself will make a lot more sense as I doubt that we will require this function for any other component.